gammapy/stats/li_ma.py

```diff
@@ -0,0 +1,347 @@
+    offset = np.full(n_on.shape, n_sig*n_sig*0.5)
+    signal = excess0.copy() + 1
+    ll = np.zeros(n_on.shape)
+    for ii in range(len(ll)):
+        ll[ii] = lm_loglikelihood(n_on[ii], n_off[ii], alpha[ii], signal[ii])
+        count = 0
+        while (l0[ii]-ll[ii]) < offset[ii] and np.isfinite(ll[ii]) and count < 100:
+            signal[ii] = (signal[ii]-excess0[ii]) * 2. + excess0[ii]
+            ll[ii] = lm_loglikelihood(n_on[ii], n_off[ii], alpha[ii], signal[ii])
+            count += 1
+
+    # Dichotomic search (i.e. interval bisection algorithm)
+    low = excess0.copy()
+    up = signal.copy()
+    signal = (up+low)/2.
+    for ii in range(len(ll)):
+        ll[ii] = lm_loglikelihood(n_on[ii], n_off[ii], alpha[ii], signal[ii])
+        count = 0.
+        while np.abs(l0[ii]-ll[ii]-offset[ii]) > 1e-4 and np.isfinite(ll[ii]) and count < 100:
+            if (l0[ii]-ll[ii]) > offset[ii]:
+                up[ii] = signal[ii]
+            else:
+                low[ii] = signal[ii]
+            signal[ii] = (up[ii]+low[ii])/2.
+            ll[ii] = lm_loglikelihood(n_on[ii], n_off[ii], alpha[ii], signal[ii])
+            count += 1
+
+    if scalar:
+        return np.abs(signal[0] - excess0[0])
+    return np.abs(signal - excess0)
+
+
+def lm_dexcess(n_on, n_off, alpha, n_sig=1.):
+    r"""Compute mean excess uncertainties of an on-off observation based on Li&Ma [1].
+
+    lm_dexcess = (lm_dexcess_up + lm_dexcess_down) / 2
+
+    Parameters
+    ----------
+    n_on : array_like
+        Observed number of counts in the on region
+    n_off : array_like
+        Observed number of counts in the off region
+    alpha : array_like
+        On / off region exposure ratio for background events
+    n_sig : float
+        Number of sigma
+
+    Returns
+    -------
+    significance : `~numpy.ndarray`
+        Error estimate
+
+    References
+    ----------
+    .. [1] Li and Ma, "Analysis methods for results in gamma-ray astronomy",
+       `Link <https://ui.adsabs.harvard.edu/abs/1983ApJ...272..317L>`_
+    """
+    err_up = lm_dexcess_up(n_on, n_off, alpha, n_sig)
+    err_down = lm_dexcess_down(n_on, n_off, alpha, n_sig)
+    err = (err_up + err_down) / 2.
+    return err
+
+
+def lm_significance_on_off(n_on, n_off, alpha):
+    r"""
+    Compute significance with the Li & Ma formula (17) [1]_.
+
+    Parameters
+    ----------
+    n_on : array_like
+        Observed number of counts in the on region
+    n_off : array_like
+        Observed number of counts in the off region
+    alpha : array_like
+        On / off region exposure ratio for background events
+
+    Returns
+    -------
+    significance : `~numpy.ndarray`
+        Significance estimate
+
+    References
+    ----------
+    .. [1] Li and Ma, "Analysis methods for results in gamma-ray astronomy",
+       `Link <https://ui.adsabs.harvard.edu/abs/1983ApJ...272..317L>`_
+    """
+    n_on = np.asanyarray(n_on, dtype=np.float64)
```

Comment:
I think in general there is not need to parse the values as arrays. Numpy takes care of this...